04-Protocol-Security.md

```diff
@@ -36,32 +36,40 @@ These primitives are chosen so that Noise Encryption layer for Stratum V2 can be
 
 #### 4.3.1.1 EC point encoding remarks
 
-Secp256k1 curve points, which includes Public Keys and ECDH results, are points with of X- and Y-coordinate, 32-bytes each. There are several possibilities how to serialize them:
+Secp256k1 curve points, i.e. Public Keys, are points with of X- and Y-coordinate.
+We serialize them in three different ways, only using the x-coordinate.
 
-1. 64-byte full X- and Y-coordinate serialization for public keys (and ECDH results) and 96 bytes for signatures.
-2. 33-byte X-coordinate with 1 parity bit serialization for public keys and similarly 65-byte for signatures.
-3. 32-byte X-coordinate only with implicit Y-coordinate for public keys and 64-byte for signatures.
+1. When signing or verifying a certificate, we use the 32 byte x-only
+encoding as defined in BIP 340.<sup>[3](#reference-3)</sup>.
 
-We choose the 32-byte serialization for public key and 64-byte for signatures with implicit Y-coordinate.
+2. When sharing keys during the handshake, whether in plain text or encrypted,
+we use the 64 byte ElligatorSwift x-only encoding as defined in BIP324<sup>[7](#reference-7)</sup> under "ElligatorSwift encoding of curve X coordinates". This encoding uses 64-bytes instead of 32-bytes in order to produce a
+pseudo-random bytesteam. This is useful because the protocol handshake starts with
+each side sending ther public key in plain text. Additionally the use of X-only
+ElligatorSwift ECDH removes the need to grind or negate private keys.
```

Comment:
This is true apparently, but in detail seems not.
When ElligatorSwift is proposed in [BIP340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#specification).
When it says "In what follows, we use the notation from BIP340", if we look at the `lift_x()` function then it seems not x-only to me.
Indeed the last step is.
`Return the unique point P such that x(P) = x and y(P) = y if y mod 2 = 0 or y(P) = p-y otherwise.`
I am not familiar with the implementation in C++ so I cannot tell where in the precise implementation this is used.